src/mbgl/renderer/layers/render_background_layer.cpp

```diff
@@ -37,8 +37,14 @@ void RenderBackgroundLayer::evaluate(const PropertyEvaluationParameters &paramet
         parameters.getCrossfadeParameters(),
         unevaluated.evaluate(parameters));
 
-    passes = properties->evaluated.get<style::BackgroundOpacity>() > 0 ? RenderPass::Translucent
-                                                                       : RenderPass::None;
+    passes = properties->evaluated.get<style::BackgroundOpacity>() == 0.0f
+        ? RenderPass::None
+        : (!unevaluated.get<style::BackgroundPattern>().isUndefined()
+           || properties->evaluated.get<style::BackgroundOpacity>() < 1.0f
+           || properties->evaluated.get<style::BackgroundColor>().a < 1.0f)
+        ? RenderPass::Translucent
```

Comment:
```
if (properties->evaluated.get<style::BackgroundOpacity>() == 0.0f) {
passes = ..
} else if (...) {
passes = ...
} else {
passes = ...
}
```

Reply:
Actually, prefer original.

Reply:
Up to you, IMO double ternary operator affects readability